edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookFrameworkServiceImpl.java

```diff
@@ -217,48 +227,51 @@ public void saveGradeMappingToGradebook(final String scaleUuid, final String gra
+                if (log.isInfoEnabled()) {
+					log.info("No Grading Scale defined yet. This is probably because you have upgraded or you are working with a new database. Default grading scales will be created. Any customized system-wide grade mappings you may have defined in previous versions will have to be reconfigured.");
+				}
                 gradingScales = GradebookFrameworkServiceImpl.this.addDefaultGradingScales(session);
             }
             return gradingScales;
         });
 	}
-	
+
 	@Override
 	public List<GradingScaleDefinition> getAvailableGradingScaleDefinitions() {
-		List<GradingScale> gradingScales = this.getAvailableGradingScales();
-		
-		List<GradingScaleDefinition> rval = new ArrayList<>();
-		for(GradingScale gradingScale: gradingScales) {
+		final List<GradingScale> gradingScales = getAvailableGradingScales();
+
+		final List<GradingScaleDefinition> rval = new ArrayList<>();
+		for(final GradingScale gradingScale: gradingScales) {
 			rval.add(gradingScale.toGradingScaleDefinition());
 		}
 		return rval;
 	}
 
-	public void setDefaultGradingScale(String uid) {
+	@Override
+	public void setDefaultGradingScale(final String uid) {
 		setPropertyValue(UID_OF_DEFAULT_GRADING_SCALE_PROPERTY, uid);
 	}
 
-	private void copyDefinitionToScale(GradingScaleDefinition bean, GradingScale gradingScale) {
+	private void copyDefinitionToScale(final GradingScaleDefinition bean, final GradingScale gradingScale) {
 		gradingScale.setUnavailable(false);
 		gradingScale.setName(bean.getName());
 		gradingScale.setGrades(bean.getGrades());
-		Map defaultBottomPercents = new HashMap();
-		Iterator gradesIter = bean.getGrades().iterator();
-		Iterator defaultBottomPercentsIter = bean.getDefaultBottomPercentsAsList().iterator();
+		final Map defaultBottomPercents = new HashMap();
+		final Iterator gradesIter = bean.getGrades().iterator();
+		final Iterator defaultBottomPercentsIter = bean.getDefaultBottomPercentsAsList().iterator();
 		while (gradesIter.hasNext() && defaultBottomPercentsIter.hasNext()) {
-			String grade = (String)gradesIter.next();
-			Double value = (Double)defaultBottomPercentsIter.next();
+			final String grade = (String)gradesIter.next();
+			final Double value = (Double)defaultBottomPercentsIter.next();
 			defaultBottomPercents.put(grade, value);
 		}
 		gradingScale.setDefaultBottomPercents(defaultBottomPercents);
 	}
 
-	private void mergeGradeMappings(Collection gradingScaleDefinitions, Session session) throws HibernateException {
-		Map newMappingDefinitionsMap = new HashMap();
-		HashSet uidsToSet = new HashSet();
-		for (Iterator iter = gradingScaleDefinitions.iterator(); iter.hasNext(); ) {
-			GradingScaleDefinition bean = (GradingScaleDefinition)iter.next();
+	private void mergeGradeMappings(final Collection gradingScaleDefinitions, final Session session) throws HibernateException {
+		final Map newMappingDefinitionsMap = new HashMap();
+		final HashSet uidsToSet = new HashSet();
```

Comment:
add generics to HashSet

Reply:
Done as park of a cleanup of that entire class